lib/token.ml

```diff
@@ -0,0 +1,12 @@
+type t =
```

Comment:
I pulled these tokens out into another module because it's the only way to refer to them outside of `lexer.ml` as say `Token.EOF` (I think)

If I put them in `lexer.mli` so that they could be used by other modules, then we wouldn't be able to use them in the implementation file without duplicating the type (and keeping that in-sync would be annoying)

In the future maybe we no longer have a need for them outside of lexer, then we can move them into `lexer.mli` as an abstract type and define them in `lexer.ml`. For now this is works and is fine